lib/cfn-nag/custom_rules/EC2NetworkAclEntryDuplicateRule.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require_relative 'base'
+
+class EC2NetworkAclEntryDuplicateRule < BaseRule
+  def rule_text
+    'A NetworkACL\'s rule numbers cannot be repeated unless one is egress and one is ingress.'
+  end
+
+  def rule_type
+    Violation::FAILING_VIOLATION
+  end
+
+  def rule_id
+    'F79'
+  end
+
+  def audit_impl(cfn_model)
+    nacl_violating_egress_entries = []
+    nacl_violating_ingress_entries = []
+    cfn_model.resources_by_type('AWS::EC2::NetworkAcl').select do |nacl|
+      nacl_violating_egress_entries += duplicate_rule_numbers?(cfn_model, nacl.network_acl_egress_entries)
+      nacl_violating_ingress_entries += duplicate_rule_numbers?(cfn_model, nacl.network_acl_ingress_entries)
+    end
+
+    nacl_violating_egress_entries.map(&:logical_resource_id) + nacl_violating_ingress_entries.map(&:logical_resource_id)
+  end
+
+  private
+
+  def duplicate_rule_numbers?(cfn_model, nacl_entries)
+    nacl_entry_resources = []
+    rule_numbers = []
+    duplicates = []
+    nacl_entries.select do |nacl_entry|
```

Comment:
i'm a bit confused.  are these select supposed to be each?  you could probably rewrite this code a lot tighter if you used select to actually select instead of using the predicate as a side effect to update the "outside" arrays

Reply:
Updated

Reply:
i think i get this logic but it took me a while of staring at the conjunction with the next and wondering about operator precedence.

the first thing i'd do is drop the ? for methods that are returning non-boolean values....

would this "one-liner" solve the problem?

nacl_entries.group_by { |entry| entry.ruleNumber }.select { |_, entries| entries.size > 1 }.map { |_, entries| entries }.flatten

Reply:
Updates are made, please re-review @erickascic when you get a chance.  Thanks!